projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/src/utils.ts

```diff
@@ -0,0 +1,254 @@
+import { translate } from './i18n';
+import { Facebook, Mail, WordPress } from './images';
+import type { UserInfo, VerbumComments } from './types';
+
+/**
+ * Returns a string of class names from the arguments.
+ * @param {...any} args - The arguments to be passed to the function.
+ */
+export function classNames( ...args: Array< string | Record< string, boolean | string > > ) {
+	const result = [];
+	for ( let i = 0; i < args.length; i++ ) {
+		if ( typeof args[ i ] === 'object' ) {
+			result[ i ] = Object.keys( args[ i ] )
+				.filter( key => args[ i ][ key ] )
+				.join( ' ' );
+		} else if ( args[ i ] ) {
+			result[ i ] = args[ i ];
+		}
+	}
+	return result.join( ' ' );
+}
+
+export const serviceData = {
+	wordpress: {
+		cookieName: 'wpc_wpc',
+		name: 'WordPress.com',
+		popup: ',height=980,width=500',
+		icon: WordPress,
+		class: 'wordpress-login',
+	},
+	facebook: {
+		cookieName: 'wpc_fbc',
+		name: 'Facebook',
+		popup: ',height=650,width=750',
+		icon: Facebook,
+		class: 'facebook-login',
+	},
+	mail: {
+		name: translate( 'Email' ),
+		icon: Mail,
+		class: 'mail-login',
+	},
+};
+
+export const canWeAccessCookies = () => {
+	// Is a WordPress cookie already set and can we read it?
+	if ( document.cookie.includes( 'wpc_' ) ) {
+		return true;
+	}
+
+	// Can we set a cookie and read our own cookie?
+	document.cookie = 'verbum_test=1; SameSite=None; Secure';
+	if ( document.cookie.includes( 'verbum_test' ) ) {
+		return true;
+	}
+
+	return false;
+};
+
+/**
+ * Uses the current bundle's size and the time it took to download and execute to estimate connection speed.
+ */
+export function isFastConnection() {
+	// Hardcoding the size of the bundle.
+	const bytes = 30000;
```

Comment:
We don't have a solution for calculating and setting this value as of now.
I removed my initial solution since it wouldn't be efficient enough. However, we still need to figure out a way to either calculate this value or another method to determine what a "fast connection" would be.

Reply:
I'm getting the same now
<img width="887" alt="Screenshot 2024-01-22 at 18 38 14" src="https://github.com/Automattic/jetpack/assets/7000684/85b1e51d-82db-4c34-bfb6-f2b288e369dc">